File: smartlist/src/main/java/com/smartlist/module/organization/dto/response/OrganizationUserRespDTO.java
package com.smartlist.module.organization.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class OrganizationUserRespDTO {

    @Schema(description = "Organization User Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer id;

    @Schema(description = "Organization name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    String name;

    @Schema(description = "Organization email", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    @Email
    String email;

    @Schema(description = "Organization role", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    String role;

    @Schema(description = "Organization department", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    String department;

    @Schema(description = "Is active", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    @NotNull
    Boolean active;

}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/organization/dto/response/organization-user.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { UserDocument } from 'src/schemas/user.schema';

export class OrganizationUserRespDto {
  @ApiProperty()
  id: string;

  @ApiProperty()
  name: string;

  @ApiProperty()
  email: string;

  @ApiProperty()
  role: string;

  @ApiProperty()
  department: string;

  @ApiProperty()
  active: boolean;

  constructor(userDocument: UserDocument) {
    this.id = userDocument.id.toString();
    this.name = userDocument.name;
    this.email = userDocument.email;
    this.role = userDocument.role;
    this.department = userDocument.department;
    this.active = userDocument.active;
  }
}
